fdbclient/ReadYourWrites.actor.cpp

```diff
@@ -1352,7 +1352,14 @@ ReadYourWritesTransaction::ReadYourWritesTransaction(Database const& cx)
     specialKeySpaceWriteMap(std::make_pair(false, Optional<Value>()), specialKeys.end), options(tr) {
 	std::copy(
 	    cx.getTransactionDefaults().begin(), cx.getTransactionDefaults().end(), std::back_inserter(persistentOptions));
-	applyPersistentOptions();
+	if (!cx->isError()) {
+		try {
+			applyPersistentOptions();
```

Comment:
This seems like it would work. If I understand correctly, at least in RYW this only affects DB defaults and not transaction persistent options, and if it were possible the most friendly way we could deal with this would be to somehow fail when setting the default. We don't currently do any validation on DB defaults, though, so we'd have to provide some mechanism to accomplish that.

There is also analogous code in the MVC layer that will probably need to be updated as well. A quick look at it made me think it is currently structured in such a way that we would run into problems with DB defaults and transaction persistent options, and I'm not sure how easy that is to work around.